public/app/features/plugins/sql/components/query-editor-raw/QueryToolbox.tsx

```diff
@@ -70,11 +71,25 @@ export function QueryToolbox({ showTools, onFormatCode, onExpand, isExpanded, ..
         <div>
           <HorizontalGroup spacing="sm">
             {onFormatCode && (
-              <IconButton onClick={onFormatCode} name="brackets-curly" size="xs" tooltip="Format query" />
+              <IconButton
+                onClick={() => {
+                  reportInteraction('grafana_sql_query_formatted', {});
```

Comment:
i think (not 100%) `validatorProps.query` exists.. could we use that to send the datasource-type here too?